shopfloor/services/zone_picking.py

```diff
@@ -940,25 +940,33 @@ def _move_line_full_qty(self, move_line, qty):
             move_line.product_uom_qty - qty, precision_rounding=rounding
         )
 
-    def _set_destination_package(self, move_line, quantity, package):
-        package_changed = False
-        response = None
+    def _is_package_not_valid(self, package, move_line):
+        message = False
         # A valid package is:
         # * an empty package
         # * not used as destination for another move line
+        # * not contains move lines with different operation type
         if not self._is_package_empty(package):
+            message = self.msg_store.package_not_empty(package)
+        elif package.planned_move_line_ids:
+            if not self.work.menu.multiple_move_single_pack:
+                message = self.msg_store.package_already_used(package)
+            elif any(
+                line.picking_id.picking_type_id != move_line.picking_id.picking_type_id
```

Comment:
maybe it is better to do it like that
```suggestion
                line.picking_id.picking_type_id.id not in self.picking_types.ids
```